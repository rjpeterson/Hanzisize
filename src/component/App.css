body {
  background-color: sandybrown;
  width: 150px;
}

.App {
  display: grid;
  width: 100%;
  grid-template-areas: 
    'logo'
    'lang'
    'tsi'
    'note'
    'error';
  align-items: center;
}

.grid-box {
  width: 100%;
  display: grid;
  align-items: center;
  justify-content: center;
}

.logo {
  grid-area: logo;
}

.box-content {
  margin: auto;
}

.logo-content img {
  height: 30px;
}

.language-input {
  grid-area: lang;
}

.text-size-input {
  grid-area: tsi;
}

.input-label {
  display: inline-flex;
}

input[type=number] {
  width: 50px;
  margin: auto;
  display: inline-flex;
  border: 2px solid grey;
  border-radius: 5px; 
}

#error-message {
  font-size: 10px;
  display: block;
}

/* .button {
  grid-area: button;
} */

.notification {
  grid-area: note;
  font-size: 10px;
  width: 90%
}

span {
  display: inline;
}

.error {
  grid-area: error;
}