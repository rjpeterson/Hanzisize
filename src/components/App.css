.error-message {
  width: 10rem;
  font-size: .9rem;
}

.App {
  display: grid;
  width: 16rem;
  grid-template-areas: 
    'top'
    'bottom';
  align-items: center;
  border-style: solid;
  border-radius: .1rem;
  border-width: .1rem;
  border-color: #5658576e;
}

.top {
  display: grid;
  grid-area: top;
  grid-template-areas: 'logo langinput';
  background-color:#5eb2ca;  
  align-items: center;
}

.bottom {
  display: grid;
  grid-area: bottom;
  grid-template-areas: 
  'mfsinput mfsinput mfsinput'
  'iframe-warning iframe-warning more-info';
  background-color: #E4E4E4;
}

.grid-box {
  width: 100%;
  display: flex;
  justify-content: center;
  margin-top: .1rem;
}

.logo {
  grid-area: logo;
  height: 2rem;
  padding: .2rem;
}