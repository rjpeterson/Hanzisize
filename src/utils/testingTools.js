/*global chrome*/

const testingTools = {
  devMode: () => {
    if(process.env.NODE_ENV === "development") {
      return true
    }
    else {
      // update_url is set by chrome webstore on submit. If it doesn't exist, the extension was loaded locally rather than installed from webstore
      return !('update_url' in chrome.runtime.getManifest());
    }
  },

  devLog: (str) => {
    if(testingTools.devMode()) {console.log(str)} 
  },

  setupTestEnv: () => {
  // npm start runs app in browser tab which doesn't have accesse to required chrome apis, so we provide them here for UI testing purposes
  if(process.env.NODE_ENV === 'development') {
    global.chrome = {
      runtime: {
        getManifest: () => {return {update_url: true}}
      },
      tabs: {
        query: ()=>{},
        sendMessage: ()=>{}
      },
      storage: {
        local: {
          get: ()=>{},
          set: ()=>{}
        }
      }
    }
  }
  }
}

export default testingTools;