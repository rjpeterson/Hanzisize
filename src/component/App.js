/*global chrome*/
import React from 'react';
import logo from '../logo.png';
import tools from '../logic/chromeStorage'
import './App.css';
import LanguageInput from './LanguageInput';
import MinFontSize from './MinFontSize';
import Notification from './Notification';
import Error from './Error';

class App extends React.Component {
  constructor(props) {
    super(props);
    const storedState = tools.getFromStorage();
    if (storedState) {this.state = {minFontSize: storedState}} else {this.state = {minFontSize: null}}
  }
  state = {
    language: null,
    validFontSize: false,
    errorMessage: null
  }

  handleLangChange = language => {
    this.setState({language: language})
  }

  handleFSChange = (valid, minFontSize) => {
    if (valid) {
      this.setState({
        minFontSize: minFontSize,
        validFontSize: true
      });
      tools.pushToStorage(minFontSize);
    } else {
      this.setState({validFontSize: false})
    }
  }

  render() {
    return (
      <div className="App">
        <header className="logo-content">
          <img src={logo} className="logo" alt="logo" />
        </header>
        <LanguageInput 
        changeHandler={this.handleLangChange}
        />
        <MinFontSize 
        changeHandler={this.handleFSChange}
        />
        <Notification 
        validFontSize={this.state.validFontSize}
        minFontSize={this.state.minFontSize}
        />
        <Error 
        message={this.errorMessage}
        />      
      </div>
    )
  };
}

export default App;
