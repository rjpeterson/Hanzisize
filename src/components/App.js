// eslint-disable-next-line no-unused-vars
/*global chrome*/

import React from 'react';
import logo from '../logo.png';
import './App.css';

import LanguageInput from './LanguageInput';
import MinFontSize from './MinFontSize';
import Notification from './Notification';
import Error from './Error';

import tools from '../logic/chromeTools';
import onAppMount from '../logic/onAppMount';

// npm start runs app in browser tab which doesn't have accesse to required chrome apis, so we provide them here for testing the UI
if(process.env.NODE_ENV === 'development') {
  global.chrome = {
    runtime: {
      getManifest: () => {return {update_url: true}}
    },
    tabs: {
      query: ()=>{}
    },
    storage: {
      local: {
        get: ()=>{},
        set: ()=>{}
      }
    }
  }
}

function isDevMode() {
  return !('update_url' in chrome.runtime.getManifest());
}

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      language: null,
      validFontSize: false,
      minFontSize: null,
      errorMessage: null,
      tabId: null,
      ready: false
    };
    
    this.handleFSChange = this.handleFSChange.bind(this);
    this.handleLangChange = this.handleLangChange.bind(this);
  }

  componentDidMount() {
    if (isDevMode()) console.log(" app.js 37 PRODUCTION MODE popup.js loaded...");

    tools.getFromStorage((storedObject) => {
      if (isDevMode()) console.log(`app.componenetDidMount storedObject: ${JSON.stringify(storedObject)}`)

      onAppMount.main((tabId) => {
        if (isDevMode()) console.log(`app.componenetDidMount tabId: ${tabId}`)

        const contentObj = {
          'language' : storedObject.language,
          'newMinFontSize': storedObject.minFontSize,
          'mode': 'initial'
        };
        
        try{tools.sendToContent(tabId, contentObj)}
        catch(err) {console.log(`app componentDidMount Could not send to content script: ${err}`)}

        this.setState({
          minFontSize: storedObject.minFontSize,
          language: storedObject.language,
          tabId: tabId,
          ready: true
        }, () => {
          if (isDevMode()) console.log(`app.componenetDidMount state: ${JSON.stringify(this.state)}`)
        });
        
      });
    })
  }

  handleLangChange(language) {
    try{tools.pushLangToStorage(language)}
    catch(err) {console.log(`Could not push to storage: ${err}`)}

    const contentObj = {
      'language' : language,
      'newMinFontSize': this.state.minFontSize,
      'mode': 'lang-change'
    };
    try{tools.sendToContent(this.state.tabId, contentObj)}
    catch(err) {console.log(`Could not send to content script: ${err}`)}

    this.setState({language: language}, () => {
      console.log('current state: ' + JSON.stringify(this.state));
      // this.setState({errorMessage: ''})
    })
  }

  handleFSChange(valid, minFontSize) {
    if (valid) {
      try{tools.pushFSToStorage(minFontSize)}
      catch(err) {console.log(`Could not push to storage: ${err}`)}

      const contentObj = {
        'language' : this.state.language,
        'newMinFontSize': minFontSize,
        'mode': 'fontsize-change'
      };
      try{tools.sendToContent(this.state.tabId, contentObj)}
      catch(err) {console.log(`Could not send to content script: ${err}`)}

      this.setState({
        minFontSize: minFontSize,
        validFontSize: true,
        errorMessage: ''
      }, () => {
        console.log('current state: ' + JSON.stringify(this.state))
      });
    } else {
      this.setState({
        validFontSize: false,
        errorMessage: 'Invalid font size'
    })
    }
  }

  render() {
    if(this.state.ready !== true) {
      return (<div>Loading...</div>)
    }
    return (
      <div className="App">
        <header className="logo-content grid-box">
          <img className="logo" src={logo} alt="logo" />
        </header>
        <LanguageInput 
        language={this.state.language}
        changeHandler={this.handleLangChange}
        />
        <MinFontSize 
        minFontSize={this.state.minFontSize}
        changeHandler={this.handleFSChange}
        />
        <Notification 
        validFontSize={this.state.validFontSize}
        minFontSize={this.state.minFontSize}
        />
        <Error 
        message={this.errorMessage}
        />      
      </div>
    )
  };
}

export default App;
