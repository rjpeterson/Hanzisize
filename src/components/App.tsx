/*global chrome*/
import React, { useState, useEffect } from "react";
import Upper from "./Upper/Upper";
import Lower from "./Lower/Lower";
import ErrorMessage from "./ErrorMessage";
import testingTools from "../utils/testingTools";
import { ContentResponse, ValidityCheck, StoredData } from "../../types";

import { makeStyles, ThemeProvider } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import theme from "./theme";
import "fontsource-roboto";

testingTools.setupTestEnv();

const useStyles = makeStyles({
  container: {
    width: 340,
    backgroundColor: theme.palette.secondary.main,
    overflow: "hidden",
    flexDirection: "row",
  },
});

function App() {
  const classes = useStyles();

  const [minFontSize, setMinFontSize] = useState<StoredData["minFontSize"]>(0);
  const [language, setLanguage] = useState<StoredData["language"]>("chinese");
  const [iFrames, setiFrames] =
    useState<ContentResponse["multipleFrames"]>(false);
  const [errorMessage, setErrorMessage] =
    useState<ValidityCheck["message"]>("");

  useEffect(() => {
    const main = () => {
      testingTools.devLog('sending "get stored data" message to background.js');

      chrome.runtime.sendMessage(
        { message: "get stored data" },
        (response: StoredData) => {
          testingTools.devLog(
            `app.js received storedData: ${JSON.stringify(
              response
            )} from background.js`
          );

          const storedMinFontSize = response.minFontSize;
          const storedLanguage = response.language;
          setMinFontSize(storedMinFontSize);
          setLanguage(storedLanguage);

          // send content object to background.js
          testingTools.devLog(
            'sending "popup opened" message to background.js'
          );
          chrome.runtime.sendMessage(
            {
              message: "popup opened",
              data: {
                minFontSize: storedMinFontSize,
                language: storedLanguage,
              },
            },
            (response) => {
              if (response) {
                // disply any url invalid or injection error messages received
                if (response.invalidUrlMessage || response.injectionError) {
                  setErrorMessage(
                    response.invalidUrlMessage || response.injectionError
                  );
                } else {
                  // display iframe warning
                  setiFrames(response.multipleFrames);
                }
              }
            }
          );
        }
      );
    };

    main();
  }, []);

  // fire resizing when user selects a new language from dropdown
  const handleLangChange = (newLanguage: string) => {
    // first store new language
    testingTools.devLog("pushing new language to storage");

    // send lang object to background.js
    chrome.runtime.sendMessage(
      { message: "handle lang change", language: newLanguage },
      (response) => {
        testingTools.devLog(`handleLangChange: ${JSON.stringify(response)}`);
      }
    );

    // finally update state
    setLanguage(newLanguage);
  };

  // fire resizing when user inputs a new minimum font size
  const handleFSChange = (minFontSize: number) => {
    // first store new minfontsize
    testingTools.devLog("pushing new font size to storage");

    chrome.runtime.sendMessage(
      { message: "handle font size change", minFontSize: minFontSize },
      (response) => {
        testingTools.devLog(`handleFSChange: ${JSON.stringify(response)}`);
      }
    );

    // finally update state
    setMinFontSize(minFontSize);
  };

  // display any error messages recieved
  if (errorMessage) {
    testingTools.devLog("display error message");
    return <ErrorMessage errorMessage={errorMessage} />;
  }
  // display active interface
  else {
    testingTools.devLog("display active interface");
    return (
      <ThemeProvider theme={theme}>
        <Container
          className={classes.container}
          disableGutters
          fixed={true}
          maxWidth="xs"
        >
          <div className="App">
            <Upper language={language} changeHandler={handleLangChange} />
            <Lower
              minFontSize={minFontSize}
              changeHandler={handleFSChange}
              iFrames={iFrames}
            />
          </div>
        </Container>
      </ThemeProvider>
    );
  }
}

export default App;
