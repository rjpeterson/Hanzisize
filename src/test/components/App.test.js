/*global chrome*/

import React from "react";
import renderer from "react-test-renderer";

import App from "../../components/App";
import testingTools from "../../utils/testingTools";

global.chrome = {
  tabs: {
    query: () => {},
  },
  storage: {
    local: {
      get: () => {},
      set: () => {},
    },
  },
};

describe("App", () => {
  it.only("renders", async () => {
    const component = renderer.create(<App />);
    await Promise.resolve();
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
