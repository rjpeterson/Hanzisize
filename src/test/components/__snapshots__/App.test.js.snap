// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="App"
>
  <header
    className="logo-container grid-box"
  >
    <div
      className="logo-background"
    >
      <img
        alt="logo"
        src="logo.png"
      />
    </div>
  </header>
  <div
    className="language-input grid-box"
  >
    <label
      className="input-label"
      htmlFor="langinput"
    >
      Select a language to resize
    </label>
    <div
      className=" css-vkzoxc-container"
      id="langinput"
      onKeyDown={[Function]}
    >
      <div
        className="language-input__control css-10k5ixo-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="language-input__value-container language-input__value-container--has-value css-1j36x9m-ValueContainer"
        >
          <div
            className="language-input__single-value css-1uccc91-singleValue"
          >
            中文
          </div>
          <div
            className="css-b8ldur-Input"
          >
            <div
              className="language-input__input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className="language-input__indicators css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className="language-input__indicator-separator css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className="language-input__indicator language-input__dropdown-indicator css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-6q0nyr-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mfs-input grid-box"
  >
    <label
      className="input-label"
      htmlFor="mfsinput"
    >
      Min Font-Size
    </label>
    <span
      className="react-numeric-input"
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "background": "#E2E2E2",
          "borderRadius": ".3rem .3rem .3rem .3rem",
          "boxShadow": "0 0 1px 1px #fff inset, 1px 1px 5px -1px #000",
          "display": "inline-block",
          "fontSize": 15,
          "padding": ".1rem .1rem .1rem .1rem",
          "position": "relative",
          "width": "7.5rem",
        }
      }
    >
      <input
        className="react-numeric-input-container"
        id="mfsinput"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        size={3}
        style={
          Object {
            "WebkitAppearance": "none",
            "border": "1px solid #ccc",
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#988869",
            "display": "block",
            "fontSize": "inherit",
            "fontWeight": 70,
            "lineHeight": "normal",
            "margin": "auto",
            "padding": ".2rem 3.4ex .2rem 3.4ex",
            "paddingLeft": " 3.4ex",
            "paddingRight": "3.4ex",
            "textAlign": "center",
            "textShadow": "1px 1px 1px rgba(0, 0, 0, 0.1)",
            "width": "100%",
          }
        }
        type="text"
        value="0"
      />
      <b
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "rgba(0,0,0,.1)",
            "borderColor": "rgba(0,0,0,.1)",
            "borderRadius": 2,
            "borderStyle": "solid",
            "borderWidth": 1,
            "bottom": 2,
            "boxShadow": "none",
            "cursor": "default",
            "position": "absolute",
            "right": 2,
            "textAlign": "center",
            "top": 2,
            "transition": "all 0.1s",
            "width": "3.3ex",
          }
        }
      >
        <i
          style={
            Object {
              "background": "rgba(0,0,0,.7)",
              "height": 2,
              "left": "50%",
              "margin": "-1px 0 0 -5px",
              "position": "absolute",
              "top": "50%",
              "width": 10,
            }
          }
        />
        <i
          style={
            Object {
              "background": "rgba(0,0,0,.7)",
              "height": 10,
              "left": "50%",
              "margin": "-5px 0 0 -1px",
              "position": "absolute",
              "top": "50%",
              "width": 2,
            }
          }
        />
      </b>
      <b
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "background": "rgba(0,0,0,.1)",
            "borderColor": "rgba(0,0,0,.1)",
            "borderRadius": 2,
            "borderStyle": "solid",
            "borderWidth": 1,
            "bottom": 2,
            "boxShadow": "none",
            "cursor": "default",
            "left": 2,
            "position": "absolute",
            "right": "auto",
            "textAlign": "center",
            "top": 2,
            "transition": "all 0.1s",
            "width": "3.3ex",
          }
        }
      >
        <i
          style={
            Object {
              "background": "rgba(0,0,0,.7)",
              "height": 2,
              "left": "50%",
              "margin": "-1px 0 0 -5px",
              "position": "absolute",
              "top": "50%",
              "width": 10,
            }
          }
        />
      </b>
    </span>
  </div>
  <div
    className="error grid-box"
  >
    <div
      className="error-content"
    >
      Please input a positive integer
    </div>
  </div>
  <div
    className="more-info-container"
  >
    <div
      className="show-more"
      onClick={[Function]}
    >
      Show More Info
    </div>
    <div
      className="inactive"
    >
      <p>
        Hi there. I'm Ryan, the sole developer of Hanzisize. I've spent hundreds of hours working on Hanzisize while making it available for anyone to use for free. If you are able, please consider donating so I can continue to improve this tool for all of us. If you're not able to donate, please leave a review of this extension so more people can find it. Thanks for your support!
      </p>
      <div
        className="coffee-button"
      >
        <a
          className="bmc-button"
          href="https://www.buymeacoffee.com/djmouthguard"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Buy me a matcha latte"
            src="bmc-new-btn-logo.svg"
          />
          <span>
            Buy me a matcha latte
          </span>
        </a>
      </div>
      <div
        className="bitcoin-info"
      >
        <p>
          or donate 
          <span
            onClick={[Function]}
          >
            Bitcoin
          </span>
        </p>
        <img
          alt="a bitcoin qr code"
          className="inactive"
          height="100"
          src="btcdonation.jpeg"
          width="100"
        />
      </div>
      <p
        className="github-info"
      >
        You can also contribute code or report bugs on 
        <a
          href="https://github.com/rjpeterson/Hanzisize"
          rel="noopener noreferrer"
          target="_blank"
        >
          Github
        </a>
      </p>
    </div>
  </div>
</div>
`;
